### PR TITLE
Deprecate USER_TOKEN_PROPERTY_NAME and add method alternatives.

### LINKED ISSUE #131: Relax visibility of CoreSocketFactory.USER_TOKEN_PROPERTY_NAME
`CoreSocketFactory.USER_TOKEN_PROPERTY_NAME` became private in #126. Would it be possible to expose it as a public constant somewhere in the library? We use it in [spring-cloud-gcp](https://github.com/spring-cloud/spring-cloud-gcp/blob/72d746486fedd19b16ead3cdbbcfb48c3e90c3a9/spring-cloud-gcp-autoconfigure/src/main/java/org/springframework/cloud/gcp/autoconfigure/sql/GcpCloudSqlAutoConfiguration.java#L210) to provide a custom application identity.

https://github.com/GoogleCloudPlatform/cloud-sql-jdbc-socket-factory/blob/74aba381161097492eaab917fc36c19a8c59df84/core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java#L617

### PR DESCRIPTION
**Pull request by kurtisvg**
Fixes #131.

### CODE REVIEW

File: core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java [hunk 1]
```diff
@@ -87,10 +87,16 @@
   public static final String MYSQL_SOCKET_FILE_FORMAT = "/cloudsql/%s";
   public static final String POSTGRES_SOCKET_FILE_FORMAT = "/cloudsql/%s/.s.PGSQL.5432";
 
+  /**
+   * Property used to set the ApplicationName for the underlying SQLAdmin client.
```
Comment (netdpb) [inline core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java]:
```suggestion
   * Property used to set the application name for the underlying SQLAdmin client.
```

File: core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java [hunk 2]
```diff
@@ -87,10 +87,16 @@
...
+   *
+   * @deprecated Replaced by {@link #getApplicationName} and {@link #setApplicationName}.
```
Comment (netdpb) [inline core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java]:
```suggestion
   * @deprecated Use {@link #setApplicationName(String)} to set the application name programmatically.
```

Most users want to set it. And it's good to use the Javadoc name with parameter types when you can.

File: core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java [hunk 3]
```diff
@@ -723,6 +728,27 @@ public SSLSocketFactory getSslSocketFactory() {
     }
   }
 
+  /**
+   * Sets the User-Agent header for requests made using the underlying SQLAdmin API client.
+   *
+   * @throws IllegalStateException if the SQLAdmin client has already been initialized.
```
Comment (netdpb) [inline core/src/main/java/com/google/cloud/sql/core/CoreSocketFactory.java]:
```suggestion
   * @throws IllegalStateException if the SQLAdmin client has already been initialized
```

File: pom.xml [hunk 4]
```diff
@@ -54,6 +54,10 @@
     </developer>
   </developers>
 
+  <properties>
```
Comment (netdpb) [inline pom.xml]: Is this supposed to be part of this PR?

Comment (kurtisvg) [reply pom.xml]: I added it to address a warning message that maven logs while compiling - I can move it out into a separate PR if you prefer.

Comment (netdpb) [reply pom.xml]: No it's fine.

**Pull request merged**